scripts/setup_irods.py

```diff
@@ -97,40 +98,65 @@ def setup_server(irods_config, json_configuration_file=None, test_mode=False):
             os.makedirs(os.path.dirname(irods_config.client_environment_path), mode=0o700)
         irods_config.commit(json_configuration_dict['service_account_environment'], irods_config.client_environment_path)
+        # password
         irods_config.admin_password = json_configuration_dict['admin_password']
-        default_resource_directory = json_configuration_dict.get('default_resource_directory', os.path.join(irods_config.irods_directory, 'Vault'))
     else:
+        # role
         determine_server_role(irods_config)
-
-        if irods_config.is_catalog:
+        # database
+        if irods_config.is_provider:
             from irods import database_interface
             l.info(irods.lib.get_header('Configuring the database communications'))
             database_interface.setup_database_config(irods_config)
-
+        # default resource
+        local_storage = determine_local_storage(irods_config)
+        if local_storage:
+            # prompt for default resource information
+            default_resource_name = get_default_resource_name(irods_config)
+            default_resource_directory = get_and_create_default_resource_vault(irods_config)
+        else:
+            # prompt and set default resource to one hosted elsewhere - the test_put will catch any typos
+            default_resource_name = get_default_resource_name(irods_config)
+            default_resource_directory = None # passed to 'setting up the database'
+        irods_config.server_config['default_resource_name'] = default_resource_name
+        # server config
         setup_server_config(irods_config)
+        # client environment and password
         setup_client_environment(irods_config)
-        default_resource_directory = get_and_create_default_vault(irods_config)
 
-    if irods_config.is_catalog:
+    if irods_config.is_provider:
         l.info(irods.lib.get_header('Setting up the database'))
-        database_interface.setup_catalog(irods_config, default_resource_directory=default_resource_directory)
+        database_interface.setup_catalog(irods_config, default_resource_directory=default_resource_directory, default_resource_name=default_resource_name)
 
     l.info(irods.lib.get_header('Starting iRODS...'))
     IrodsController(irods_config).start(test_mode=test_mode)
 
-    if irods_config.is_resource:
+    # create local storage resource for consumer (provider was configured directly in database_interface.setup_catalog above)
+    if irods_config.is_consumer:
         irods.lib.execute_command(['iadmin', 'mkresc', irods_config.server_config['default_resource_name'], 'unixfilesystem', ':'.join([irods.lib.get_hostname(), default_resource_directory]), ''])
 
+    # update core.re with default resource
+    core_re_path = os.path.join(irods_config.core_re_directory, 'core.re')
+    replace_in_file(core_re_path, 'demoResc', irods_config.server_config['default_resource_name'])
```

Comment:
will only work once, on initial run of setup_irods.py.

i think this is a feature, not a bug.